payjoin/src/receive/mod.rs

```diff
@@ -491,7 +491,8 @@ impl MaybeInputsSeen {
     pub fn check_no_inputs_seen_before(
         self,
         is_known: impl Fn(&OutPoint) -> Result<bool, Error>,
-    ) -> Result<OutputsUnknown, Error> {
+        is_receiver_output: impl Fn(&Script) -> Result<bool, Error>,
+    ) -> Result<ProvisionalProposal, Error> {
```

Comment:
I can see why these might be merged, but perhaps the checks should return `RequestError` (since if they fail, the request is bad) while the closure returns Error, since that's a problem with the server implementation. Either way this is separate from the ProvisionalProposal change